### PR TITLE
D game stats

### PR DESCRIPTION
**Pull request by DavidTTran**
Completed

- Iteration 1's game_stats methods located in `stat_tracker.rb`
- Tests for iteration 1 and changed returns values to match the spec harness seasons as a String, and percentages as a decimal.
- Refactored majority of the methods to use 2 separate helper methods. (`sort`, and `games_outcome_percent`)
- Iteration 4's most/least accurate, most/fewest tackles methods and accompanying tests.
- Moved `sort`, `games_outcome_percent`, `games_in_a_season` into the Game class as class methods.
- 14 methods, take approximately 14 seconds to run through the spec harness.

To do...

- Create either an inherited class or modules, to refactor the StatTracker
- Merge each iteration into dev

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -13,6 +13,38 @@ def self.games=(value)
...
+      elsif outcome == "home" && value.home_goals > value.away_goals
+        games_count += 1
+      elsif outcome == "draw" && value.home_goals == value.away_goals
+        games_count += 1
+      end
+    end
+    (games_count / Game.all.length).round(2)
+  end
+
+  def self.games_in_a_season(season)
```
Comment (brian-greeson) [inline lib/game.rb:42]: I like these helper methods living inside the game class

File: lib/game.rb [hunk 2]
```diff
@@ -26,7 +58,7 @@ def self.games=(value)
 
   def initialize(data)
     @game_id = data[:game_id]
-    @season = data[:season]
+    @season = data[:season].to_s
```
Comment (brian-greeson) [inline lib/game.rb:61]: Why did you need to turn this into a string?

File: lib/game_team.rb [hunk 3]
```diff
@@ -14,6 +14,12 @@ def self.game_teams=(value)
     @@game_teams = value
   end
 
+  def self.season_games(games)
```
Comment (brian-greeson) [inline lib/game_team.rb:17]: Am I correct in assuming the parameter `games` is a collection of all games you want to find matching gameteam entries for?

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -18,4 +24,138 @@ def self.create_items(file, item_class)
       CSV.foreach(file, csv_options) { |row| item_class.add(item_class.new(row.to_hash)) }
   end
 
+  def highest_total_score
```
Comment (brian-greeson) [inline lib/stat_tracker.rb:27]: Very readable functions!

**Review by brian-greeson — APPROVED** [review]
Great Job! I like your helper methods being accessible from the class files. I think on the next refactor we can improve some naming.  Overall looks great.

File: lib/game.rb (hunk 2 above)
Comment (DavidTTran) [reply lib/game.rb:61]: I think whenever the spec harness asks for a season as a return value it asks for the `season` to be represented as a `String` for example;

```
{
"20122013" => 3.86,
"20162017" => 4.75,
"20142015" => 4.00,
"20152016" => 3.88,
"20132014" => 4.33
}

File: lib/game_team.rb (hunk 3 above)
Comment (DavidTTran) [reply lib/game_team.rb:17]: It's similar to your methods that you showed on slack. The `Game - games_in_a_season` return an array of game objects that have a similar season as the parameter. `GameTeam - season_games` takes the return value of `games_in_a_season` and through `GameTeam.all` for matching `game_ids`

**Comment by kristastadler** [comment]
Did group pull request review. Decided to change sort method name to find_all_scores. Otherwise we were all in agreement on this pull request and are merging it to dev.
